data_retriever/migration_event_queue.py

```
@@ -0,0 +1,204 @@
+            self._generate_migration_id()
+            if self._migration_id == "":
+                raise EventQueueException("No migration has been started")
+        migration_id = "migration_" + self._migration_id
+        try:
+            self._cursor.execute("""
+                SELECT "action", "metadata" FROM "history_event" WHERE "entityId"=%s ORDER BY "createdAt" DESC
+            """, (migration_id,))
+            rows = self._cursor.fetchall()
+            return [deserialize_event(row[0], row[1]) for row in rows]
+        except Exception as e:
+            raise EventQueueException(f"Failed to get events from Redis: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)